QuickFIXn/SocketReader.cs

```diff
@@ -200,6 +202,13 @@ protected void ProcessStream()
 
         protected void DisconnectClient()
         {
+            _readCancellationTokenSource.Cancel();
```

Comment:
It feels like these lines should be in `Dispose` and `DisconnectClient` should just call `Dispose`